ecl/hql/hqlparse.cpp

```diff
@@ -568,6 +569,52 @@ void HqlLex::pushMacro(IHqlExpression *expr)
     }
 }
 
+void HqlLex::checkSignature()
+{
+    YYSTYPE dummyToken;
+    dummyToken.setPosition(0,0,0,text->querySourcePath());
+    Owned<IPipeProcess> pipe = createPipeProcess();
+    pipe->run("gpg", "gpg --verify -", ".", true, false, true, 0, false);
```

Comment:
Is there any chance that this can be subverted by replacing gpg?  Does it need a full path?  Probably if that can happen all bets are off anyway.